src/Symfony/Component/Config/composer.json

```diff
@@ -17,7 +17,8 @@
     ],
     "require": {
         "php": ">=5.5.9",
-        "symfony/filesystem": "~2.8|~3.0"
+        "symfony/filesystem": "~2.8|~3.0",
+        "symfony/dependency-injection": "~3.2"
```

Comment:
IMO that's not a hard dependency as the Config component itself does not require the DependencyInjection component to be installed.